### ISSUE TITLE
Query on field non unicode string should not append N

### ISSUE THREAD

**Issue by fchiumeo** [aspnet/EntityFrameworkCore#9582]
I have setting all string use non unicode varchar and select append N in generated SQL

### Steps to reproduce
```c#
DbContext
protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            foreach (var property in modelBuilder.Model.GetEntityTypes().SelectMany(e => e.GetProperties().Where(p => p.ClrType == typeof(string))))
            {
                property.IsUnicode(false);
            }
        }
```
```c#
 public class TipoServicioConfig : IEntityTypeConfiguration<TipoServicio>
    {
        public void Configure(EntityTypeBuilder<TipoServicio> builder)
        {
            builder.HasKey(ts => ts.Id);

            builder.Property(ts => ts.Id).IsRequired();
            builder.Property(ts => ts.Nombre).IsRequired().HasMaxLength(20);
            
           //it does not work either
          //builder.Property(ts => ts.Nombre).IsRequired().HasMaxLength(20).HasColumnType("varchar(20)");
        }
    }
```

```c#
   public class TipoServicio
    {
 
        public int Id{ get; set; }
        public string Nombre { get; set; }

    }
```
```c#
var query = DbContext.Set<TipoServicio>().Where(xx => xx.Nombre.Contains("lla")).ToList();
```

```sql
      SELECT [xx].[Id], [xx].[Nombre]
      FROM [TipoServicio] AS [xx]
      WHERE CHARINDEX(N'lla', [xx].[Nombre]) > 0
```

### Further technical details
EF Core version: 2.0.0
Database Provider:  Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017 (15.3)

**Comment by smitpatel** [comment]
See https://github.com/aspnet/EntityFrameworkCore/issues/4978#issuecomment-298977909

Duplicate of #4978

**Comment by smitpatel** [comment]
Keeping this open to discuss  if we want to add `AsUnicode` `AsNonUnicode` API.

**Comment by fchiumeo** [comment]
please add ````AsNonUnicode```` Api or mapping correctly

**Comment by ajcvickers** [comment]
Triage: try to fix just the Contains issue for now.

**Comment by smitpatel** [comment]
Blocked on #13192

**Comment by justintubbs** [comment]
For those using 2.2 (or pre 3.0) one way to solve this for Oracle is to use a static string value (which will get converted to a parameter) and will no longer have the `N` prefix in front:

Linq: (.ComorbidDetail is a nullable column of varchar(100) in Oracle table)
```
string staticValue = "0x00D7"; // Whatever you want here
var query = from c in _dbContext.MemComorbids.AsNoTracking()        
	join cc in _dbContext.ComorbidCrosswalks
	on new
	{
		c.ComorbidCode,
		ComorbidDetail = c.ComorbidDetail == null ? staticValue : c.ComorbidDetail
	} equals new
	{
		cc.ComorbidCode,
		ComorbidDetail = cc.ComorbidDetail == null ? staticValue : cc.ComorbidDetail
	} into c_cc
	from crosswalk in c_cc.DefaultIfEmpty()
	select c;
```
Oracle SQL Query:
```
SELECT *
FROM "SCHEMA"."PATIENT_COMORBIDS" "c0"
LEFT JOIN "SCHEMA"."COMORBID_CROSSWALK" "cc0" ON ("c0"."CODE" = "cc0"."CMBD_CODE") AND (CASE
    WHEN "c0"."DETAIL" IS NULL
    THEN :staticValue_0 ELSE "c0"."DETAIL"
END = CASE
    WHEN "cc0"."CMBD_DETAIL" IS NULL
    THEN :staticValue_0 ELSE "cc0"."CMBD_DETAIL"
END)
```
